streaming/replication.c

```diff
@@ -0,0 +1,1931 @@
+                // This is safe but do not do this from production code. (The debugging points that call this are
+                // on the collector thread and this is the hot-page so it cannot be flushed during execution).
+                uv_rwlock_rdlock(&((struct rrdeng_collect_handle *)rd->state->handle)->ctx->pg_cache.pg_cache_rwlock);
+
+                struct rrdeng_page_descr *descr = ((struct rrdeng_collect_handle *)rd->state->handle)->descr;
+                struct page_cache_descr *pc_descr = descr->pg_cache_descr;
+                if (pc_descr) {
+                    storage_number x;
+                    uint32_t entries = descr->page_length / sizeof(x);
+                    uint32_t start = 0;
+                    if (entries > 3)
+                        start = entries-3;
+                    char buffer[80] = {0};
+                    for(uint32_t i=start; i<entries; i++) {
+                        x = ((storage_number*)pc_descr->page)[i];
+                        sprintf(buffer + strlen(buffer), STORAGE_NUMBER_FORMAT " ", x);
+                    }
+                    debug(D_REPLICATION, "%s.%s page_descr %llu - %llu with %u, last points %s", st->id, rd->id,
+                                         descr->start_time,
+                                         descr->end_time,
+                                         entries,
+                                         buffer);
+                }
+                uv_rwlock_rdunlock(&((struct rrdeng_collect_handle *)rd->state->handle)->ctx->pg_cache.pg_cache_rwlock);
+            }
+            #endif
+        }
+        netdata_rwlock_unlock(&st->rrdset_rwlock);
+    }
+#else
+    UNUSED(st);
+#endif
+}
+
+void print_collected_metric_past_data(RRDDIM_PAST_DATA *past_data, REPLICATION_STATE *rep_state){
+
+    RRDSET *st = rrdset_find_byname(rep_state->host, past_data->rrdset_id);
+    if(unlikely(!st)) {
+        error("Cannot find chart with name_id '%s' on host '%s'.", past_data->rrdset_id, rep_state->host->hostname);
+        return;
+    }
+    past_data->rd = rrddim_find(st, past_data->rrddim_id);
+    if(unlikely(!past_data->rd)) {
+        error("Cannot find dimension with id '%s' in chart '%s' on host '%s'.", past_data->rrddim_id, past_data->rrdset_id, rep_state->host->hostname);
+        return;
+    }
+
+    RRDDIM *rd = past_data->rd;
+    time_t ts = past_data->start_time  / USEC_PER_SEC;
+    time_t te = past_data->end_time  / USEC_PER_SEC;
+    storage_number *page = (storage_number *)past_data->page;
+    uint32_t len = past_data->page_length / sizeof(storage_number);
+    
+    info("%s: Past Samples(%u) [%ld, %ld] for dimension %s\n", REPLICATION_MSG, len, ts, te, rd->id);
+    time_t t = ts;
+    for(uint32_t i=0; i < len ; i++){
+        info("T: %ld, V: "STORAGE_NUMBER_FORMAT" \n", t, page[i]);
+        t += rd->update_every;
+    }
+}
+
+unsigned int is_localhost(RRDHOST* host){
```

Comment:
This function has potential to be used in other places of our code, it should not be declared here inside `replication` instead it should be inside `rrdhost.c`.